x/smart/stream_dialer.go

```diff
@@ -372,17 +393,21 @@ func (f *StrategyFinder) findFallback(ctx context.Context, testDomains []string,
 				psiphonCfg := v.Psiphon
 				psiphonJSON, err := json.Marshal(psiphonCfg)
 				if err != nil {
-					f.logCtx(ctx, "Error marshaling to JSON: %v, %v", psiphonCfg, err)
+					f.logCtx(ctx, "Error marshaling to JSON: %v, %v\n", psiphonCfg, err)
 				}
+				psiphonSignature := f.getPsiphonConfigSignature(psiphonJSON)
 
-				dialer, err := newPsiphonDialer(ctx, psiphonJSON)
+				dialer, err := newPsiphonDialer(f, ctx, psiphonJSON, psiphonSignature)
 				if err != nil {
 					return nil, fmt.Errorf("getPsiphonDialer failed: %w", err)
 				}
 
-				// TODO(laplante): test the psiphon dialer
+				err = f.testDialer(raceCtx, dialer, testDomains, psiphonSignature)
```

Comment:
Let's move the testDialer call outside the `switch`. The `switch` should be for constructing the dialer only (consider moving it to a separate function).

Then, can we have the error logging in where we call testDialer?